src/BotService/Infrastructure/Core/CallHandler.cs

```diff
@@ -275,6 +288,85 @@ private static StartStreamExtractionResponse GetStreamExtractionResponse(Protoco
             }
         }
 
+        private int AssignExtractionPort(StartStreamExtractionBody streamBody)
+        {
+            int port = 0;
+
+            switch (streamBody.Protocol)
+            {
+                case Protocol.SRT:
+                    if (!_availablePorts.TryTake(out int srtPort))
+                    {
+                        _logger.LogWarning("[Call Handler] There are no ports available");
```

Comment:
**suggestion(non-blocking):**We can add the protocol to the error message. E.g: There are no SRT ports available. Same for RTMP and RTMPS messages